src/variables/tree.tsx

```diff
@@ -42,12 +42,21 @@ export class VariablesBodyTree extends ReactWidget {
             key={scope.name}
             service={this._service}
             data={scope.variables}
+            filter={this._filter}
           />
         ))}
       </>
     );
   }
 
+  /**
+   * Set the variable filter list.
+   */
+  set filter(filter: Set<string>) {
+    this._filter = filter;
+    this.update();
+  }
```

Comment:
The table and tree viewers could benefit from some refactoring, to avoid having this logic at 2 different places.